sormas-ui/src/main/java/de/symeda/sormas/ui/utils/ArchiveController.java

```diff
@@ -0,0 +1,250 @@
+				I18nProperties.getString(Strings.yes),
+				I18nProperties.getString(Strings.no),
+				null,
+				e -> {
+					if (Boolean.TRUE.equals(e)) {
+						Date endOfProcessingDate = entityFacade.calculateEndOfProcessingDate(entityUuids);
+						entityFacade.archiveCoreEntities(entityUuids, endOfProcessingDate);
+
+						callback.run();
+						new Notification(
+							I18nProperties.getString(archivedHeading),
+							I18nProperties.getString(archivedMessage),
+							Notification.Type.HUMANIZED_MESSAGE,
+							false).show(Page.getCurrent());
+					}
+				});
+		}
+	}
+
+	public void dearchiveSelectedItems(
+		List<String> entityUuids,
+		CoreFacade entityFacade,
+		String noSelectionMessage,
+		String messageNoEntitySelected,
+		String dearchiveConfirmationMessage,
+		String entity,
+		String headingConfirmationDeachiving,
+		String headingEntityDearchived,
+		String messageEntityDearchived,
+		Runnable callback) {
+
+		if (entityUuids.isEmpty()) {
+			new Notification(
+				I18nProperties.getString(noSelectionMessage),
+				I18nProperties.getString(messageNoEntitySelected),
+				Notification.Type.WARNING_MESSAGE,
+				false).show(Page.getCurrent());
+		} else {
+			VerticalLayout verticalLayout = new VerticalLayout();
+
+			Label contentLabel = new Label(
+				String.format(
+					String.format(I18nProperties.getString(dearchiveConfirmationMessage), entityUuids.size()),
+					I18nProperties.getString(entity).toLowerCase(),
+					I18nProperties.getString(entity).toLowerCase()));
+			contentLabel.setWidth(100, Sizeable.Unit.PERCENTAGE);
+			verticalLayout.addComponent(contentLabel);
+
+			TextArea dearchiveReason = new TextArea();
+			dearchiveReason.setCaption(I18nProperties.getCaption(Captions.dearchiveReason));
+			dearchiveReason.setWidth(100, Sizeable.Unit.PERCENTAGE);
+			dearchiveReason.setRows(2);
+			dearchiveReason.setRequiredIndicatorVisible(true);
+			verticalLayout.addComponent(dearchiveReason);
+
+			VaadinUiUtil.showConfirmationPopup(
+				I18nProperties.getString(headingConfirmationDeachiving),
+				verticalLayout,
+				I18nProperties.getString(Strings.yes),
+				I18nProperties.getString(Strings.no),
+				null,
+				connfirmed -> {
+					if (Boolean.TRUE.equals(connfirmed)) {
+						if (dearchiveReason.getValue().isEmpty()) {
+							dearchiveReason.setComponentError(new UserError(I18nProperties.getString(Strings.messageArchiveUndoneReasonMandatory)));
+							return false;
+						}
+						entityFacade.dearchiveCoreEntities(entityUuids, dearchiveReason.getValue());
+
+						callback.run();
+						new Notification(
+							I18nProperties.getString(headingEntityDearchived),
+							I18nProperties.getString(messageEntityDearchived),
+							Notification.Type.HUMANIZED_MESSAGE,
+							false).show(Page.getCurrent());
+					}
+					return true;
+				});
+		}
+	}
+
+	public void navigateToView(String viewName, String caseUuid, ViewMode viewMode) {
```

Comment:
Please remove these `navigateToView` and `navigateToData` methods that are specific to certain entities here, and rather pass them in as callbacks if you need them here; the logic should stay in the CaseController and EventController.

Reply:
removed from all core controllers